libsrc/apple2/opendir.s

```diff
@@ -0,0 +1,156 @@
+;
+; Colin Leroy-Mira <colin@colino.net>, 2024
+;
+; DIR* __fastcall__ opendir (register const char* name)
+;
+
+        .export   _opendir
+
+        .import   _open, _read, _close
+        .import   _malloc, _free
+        .import    ___directerrno
+
+        .import   ___oserror, __cwd
+
+        .import   pushptr1, popptr1
+        .import   pushax, pusha0
+
+        .importzp ptr1
+
+        .include  "apple2.inc"
+        .include  "dir.inc"
+        .include  "errno.inc"
+        .include  "fcntl.inc"
+        .include  "zeropage.inc"
+
+.proc _opendir
+        sta       ptr1
+        stx       ptr1+1
+
+        ldy       #$00
+        lda       (ptr1),y
+        cmp       #'.'
+        bne       :+
+
+        lda       #<__cwd
+        ldx       #>__cwd
+        sta       ptr1
+        stx       ptr1+1
+
+:       ; open directory
+        jsr       pushptr1
+        lda       #O_RDONLY
+        jsr       pusha0
+
+        ldy       #$04
+        jsr       _open
+
+        cmp       #$FF        ; Did we succeed?
+        beq       @return_null
+        pha                   ; Yes - Push fd for backup
+
+        ; malloc the dir struct
+        lda       #<.sizeof(DIR)
+        ldx       #>.sizeof(DIR)
+        jsr       _malloc
+        bne       :+
+
+        ; We failed to allocate
+        pla                   ; Get fd back
+        ldx       #$00
+        jsr       _close      ; close it
```

Comment:
Start comment in uppercase.

Reply:
As this one is merged, addressed in #2535